### PR TITLE
[Inductor] Convert 0d CPU tensor to scalar during triton codegen

### PR DESCRIPTION
**Pull request by yanboliang**
This is a follow up to address [this](https://github.com/pytorch/torchdynamo/pull/1284#pullrequestreview-1130319129). We revised to use the codegen approach to handle 0d CPU tensor, which will not support cudagraph any more.

cc @jansel @lezcano @fdrocha @mlazos @soumith @voznesenskym

### CODE REVIEW

File: torch/_inductor/codegen/triton.py [hunk 1]
```diff
@@ -791,9 +800,13 @@ def load(self, name: str, index: sympy.Expr):
             other = ", other=0"
         else:
             other = ""
-        line = f"tl.load({var} + ({index}), {mask}{ep}{other})"
-        if V.graph.get_dtype(name) in (torch.float16, torch.bfloat16):
-            line += ".to(tl.float32)"
+
+        if V.graph.is_unspec_arg(name):
+            line = var
```
Comment (ngimel) [inline torch/_inductor/codegen/triton.py:805]: can you handle conversion to fp32 here instead of type annotation?

**Comment by ngimel** [comment]
Does this fix https://github.com/pytorch/torchdynamo/issues/1681#issuecomment-1283433527? Can you add a test with a similar pattern, a mix of cpu and cuda ops?

**Review by ngimel — APPROVED** [review]

**Comment by yanboliang** [comment]
> Does this fix https://github.com/pytorch/torchdynamo/issues/1681#issuecomment-1283433527? Can you add a test with a similar pattern, a mix of cpu and cuda ops?

Yes, it fixed.

**Pull request closed without merging**